framework/src/functions/PiecewiseLinearBase.C

```diff
@@ -43,51 +47,62 @@ PiecewiseLinearBase::buildInterpolation(const bool extrap)
   }
 }
 
+template <typename BaseClass>
 Real
-PiecewiseLinearBase::value(Real t, const Point & p) const
+PiecewiseLinearBaseTempl<BaseClass>::value(Real t, const Point & p) const
 {
   const auto x = _has_axis ? p(_axis) : t;
   return _scale_factor * _linear_interp->sample(x);
 }
 
+template <typename BaseClass>
 ADReal
-PiecewiseLinearBase::value(const ADReal & t, const ADPoint & p) const
+PiecewiseLinearBaseTempl<BaseClass>::value(const ADReal & t, const ADPoint & p) const
```

Comment:
so the non AD version also has this member?

Reply:
yes. This is to allow "hacks" for people like @joshuahansel who want to pass variables (evaluated at quadrature points) in as the `t` argument (and I think I've actually seen people pass vector-like variables in for `p` as well) and get AD operations

Reply:
can't @joshuahansel use ADFunctions for this though?
Are ADFunctions only a thing for our requirement to get a functor returning ADReal?

Reply:
let's resolve this for the merge and discuss later